SAMv1.tex

```diff
@@ -307,6 +309,25 @@ \subsection{The header section}
 \end{longtable}
 \end{center}
 
+\subsubsection{Defined sub-sort terms}\label{sec:sub-sort}
+
+While sub-sort permits implementation dependent sorting, some terms are defined to have specific meanings.
+
+\begin{description}
+\item[lexicographical] sort order is defined as a character based dictionary sort with the character order as defined by the POSIX C locale.
+  For example ``abc'', ``abc17'', ``abc5'', ``abc59'' and ``abcd'' are in lexicographical order.
+
+\item[natural] sort order is an ordering of strings in the lexicographical order, except that multi-digit natural numbers (positive integers excluding leading zeros) are ordered as a single character (collating immediately after the single-digit numbers).
+  For example, ``abc-5'', ``abc'', ``abc05'', ``abc5'', ``abc17'', ``abc59'' and ``abcd'' are in natural order.
```

Comment:
add "abb" and "abd" and "abc+5" "abc-6" "abc/" and "abc:" as well.

Reply:
Need to specify that leading zeros are Tiebroken by lexicographic order so that "abc05" "abc5" "abc05" is not properly sorted. (unlike SamTools now.)

Reply:
Decimal fractions won't be very natural! Consider 6.1, 6.05, 6.12 which would be in "natural" order!

Reply:
I'm wary of creating an example list which is so vast it goes on for ages!  However agreed we can add more and fractions are definitely worth putting in the examples.

For what it's worth, the text DOES already cover minuses, fractions and leading zeros by stating "_multi-digit natural numbers (positive integers excluding leading zeros)_".  If people read the spec then it's clear -050.5 is - 0 50 . 5 with two natural numbers in there for purpose of sorting (50 and 5).

Reply:
Ok maybe not that clear - samtools actually works differently to how I expected.  It does use leading zeros in the calculation, with ties for the same equality of value being those with the most leading zeroes come first.

Thus this is ordered:  abc03 abc5 abc008 abc08 abc8 abc17